zcash_client_sqlite/src/wallet.rs

```diff
@@ -129,6 +131,32 @@ pub(crate) mod scanning;
 
 pub(crate) const BLOCK_SAPLING_FRONTIER_ABSENT: &[u8] = &[0x0];
 
+enum AccountTypes {
```

Comment:
I'm renaming to singular as suggested. 
I do not plan on using different flags for different imported keys. The flag distinguishes zip32 from non-zip32. UFVK and UIVK can both be represented in the same UVK column which I renamed for that purpose.
With ZIP-316 rev. 1 allowing UVKs to contain only a transparent key, we can store even more account types in the table.
The only thing our present schema cannot handle I guess would be the non-HD keys (e.g. the private key to a specific T address).

Reply:
The problem with using the same flag for imported UIVKs and imported UFVKs is that this means that you have to actually deserialize the UVK to know what sort it is; this causes a couple of problems, but the main one I'm concerned about is that it means that you can't use the account type to filter in SQL queries, which is necessary for example for note selection (we can't select notes from IVK-only accounts, because we can't tell whether they're unspent.)

Reply:
Makes sense. Enum renamed.